src/mesh/stencils_sympy.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+
+from sympy import Symbol, Eq
+from sympy.matrices import Matrix
+from sympy.printing import ccode
+from sympy.simplify import combsimp as simp
+from sympy.utilities.codegen import codegen
+
+
+def pow(a,b):
+    if b == 0:
+        return "1"
+    if b == 1:
+        return a
+    else:
+        return "%s**%d"%(a,b)
+
+def fac(a):
+    if a==0 or a==1:
+        return 1
+    else:
+        assert(a>0)
+        return a*fac(a-1)
+
+def gen_code(order,matrix):
+    global x
+    x=[]
+    for i in range(order):
+        x.append(Symbol("spacing.f%d"%i))
+    A=Matrix(order,order,matrix)
+
+    try:
+        iA=A.inv()
+    except:
```

Comment:
Why not?

from [here](https://www.tutorialspoint.com/python3/python_exceptions.htm)
This kind of a try-except statement catches all the exceptions that occur. Using this kind of try-except statement is not considered a good programming practice though, because it catches all exceptions but does not make the programmer identify the root cause of the problem that may occur.

However, the exception is raised as is. This is not worse then not try/except.
It however gives some infos, that may or may not be useful.